zstacklib/zstacklib/utils/upload_task.py

```
@@ -0,0 +1,311 @@
+            if not p.filename:
+                continue
+
+            logger.debug("uploading image %s: %s slice, index: %d, offset: %d, content length: %d" %
+                         (param.task_uuid, p.filename, param.slice_index, param.slice_offset, param.slice_size))
+
+            slice_downloaded_size = 0
+            try:
+                reader = SizedReader(p.fp, None, param.slice_offset)
+                remaining = param.slice_size
+                bytes_read = 0
+                md5 = hashlib.md5()
```

Ruff (0.14.0)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)